herddb-core/src/main/java/herddb/sql/expressions/SQLParserExpressionCompiler.java

```diff
@@ -0,0 +1,163 @@
+/*
+ Licensed to Diennea S.r.l. under one
+ or more contributor license agreements. See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership. Diennea S.r.l. licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+
+ */
+
+package herddb.sql.expressions;
+
+import herddb.model.StatementExecutionException;
+import herddb.sql.DDLSQLPlanner;
+import net.sf.jsqlparser.expression.Expression;
+import net.sf.jsqlparser.expression.JdbcParameter;
+import net.sf.jsqlparser.expression.LongValue;
+import net.sf.jsqlparser.expression.StringValue;
+import net.sf.jsqlparser.expression.TimestampValue;
+
+/**
+ * Created a pure Java implementation of the expression which represents the given jSQLParser Expression
+ *
+ * @author enrico.olivelli
+ */
+public class SQLParserExpressionCompiler {
+
+    public static CompiledSQLExpression compileExpression(Expression expression) {
+        if (expression == null) {
+            return null;
+        }
+        if (expression instanceof JdbcParameter) {
+            JdbcParameter p = (JdbcParameter) expression;
+            return new JdbcParameterExpression(p.getIndex() - 1);
+        } else if (expression instanceof StringValue
+                || expression instanceof LongValue
+                || expression instanceof TimestampValue) {
+            return new ConstantExpression(DDLSQLPlanner.resolveValue(expression, false));
+        }
+//        else if (expression instanceof RexLiteral) {
```

Comment:
TODO: implement all Expression types, I am leaving Calcite code, as far as test fail I am going to implement all of the supported expressions and operators